freshmaker/kojiservice.py

```diff
@@ -407,6 +412,29 @@ def get_bundle_csv(self, build_nvr):
         with zipfile.open(csv_files[0]) as f:
             return yaml.safe_load(f)
 
+    @region.cache_on_arguments()
+    def get_modulemd(self, build_nvr):
+        """
+        Get Modulemd.ModuleStreamV2 instance through build NVR
+        which can be used to get module's name, stream, etc.
+
+        :param str build_nvr: NVR of module build.
+        :return: Modulemd.ModuleStreamV2 instance.
+        """
+        mmd = None
+        build = self.get_build(build_nvr)
+        try:
+            modulemd_str = build["extra"]["typeinfo"]["module"]["modulemd_str"]
+        except KeyError:
+            return None
+
+        try:
+            _, mmd = Modulemd.read_packager_string(modulemd_str, None, None)
```

Comment:
I'm not familiar with Modulemd, could you explain it a bit to me? Thanks.

Reply:
This is a library to manipulate module metadata files. More detail, you can check https://github.com/fedora-modularity/libmodulemd
I am using this api https://fedora-modularity.github.io/libmodulemd/latest/modulemd-2.0-Modulemd.html#modulemd-read-packager-string   to parse the module's name and stream from a yaml string.

Reply:
Thank you!